### ISSUE TITLE
Unable to add PFSense

### ISSUE THREAD

**Issue by paulharman** [xpipe-io/xpipe#153]
Hi,

I'm struggling to add PFsense as a connection.  When remoting on to it you don't get a shell but rather a menu system.  Any ideas how I can add this?

Here's the output of the add window

`Shell did not start up in time:
Netgate pfSense Plus - Serial: xxxx - Netgate Device ID: xxxx

*** Welcome to Netgate pfSense Plus 23.01-RELEASE (amd64) on pfSense ***

 WAN (wan)       -> igb1       -> v4/DHCP4: x.x.x.x/20
 LAN (lan)       -> igb0       -> v4: 192.168.0.1/24
 OPT1 (opt1)     -> em0        -> 

 0) Logout (SSH only)                  9) pfTop
 1) Assign Interfaces                 10) Filter Logs
 2) Set interface(s) IP address       11) Restart webConfigurator
 3) Reset webConfigurator password    12) PHP shell + Netgate pfSense Plus tools
 4) Reset to factory defaults         13) Update from console
 5) Reboot system                     14) Disable Secure Shell (sshd)
 6) Halt system                       15) Restore recent configuration
 7) Ping host                         16) Restart PHP-FPM
 8) Shell

Netgate pfSense Plus - Serial: xxxx - Netgate Device ID: xxxx

*** Welcome to Netgate pfSense Plus 23.01-RELEASE (amd64) on pfSense ***

 WAN (wan)       -> igb1       -> v4/DHCP4: x.x.x.x/20
 LAN (lan)       -> igb0       -> v4: 192.168.0.1/24
 OPT1 (opt1)     -> em0        -> 

 0) Logout (SSH only)                  9) pfTop
 1) Assign Interfaces                 10) Filter Logs
 2) Set interface(s) IP address       11) Restart webConfigurator
 3) Reset webConfigurator password    12) PHP shell + Netgate pfSense Plus tools
 4) Reset to factory defaults         13) Update from console
 5) Reboot system                     14) Disable Secure Shell (sshd)
 6) Halt system                       15) Restore recent configuration
 7) Ping host                         16) Restart PHP-FPM
 8) Shell

Netgate pfSense Plus - Serial: xxxx - Netgate Device ID: xxxx

*** Welcome to Netgate pfSense Plus 23.01-RELEASE (amd64) on pfSense ***

 WAN (wan)       -> igb1       -> v4/DHCP4: x.x.x.x/20
 LAN (lan)       -> igb0       -> v4: 192.168.0.1/24
 OPT1 (opt1)     -> em0        -> 

 0) Logout (SSH only)                  9) pfTop
 1) Assign Interfaces                 10) Filter Logs
 2) Set interface(s) IP address       11) Restart webConfigurator
 3) Reset webConfigurator password    12) PHP shell + Netgate pfSense Plus tools
 4) Reset to factory defaults         13) Update from console
 5) Reboot system                     14) Disable Secure Shell (sshd)
 6) Halt system                       15) Restore recent configuration
 7) Ping host                         16) Restart PHP-FPM
 8) Shell

Netgate pfSense Plus - Serial: xxxx - Netgate Device ID: xxxx

*** Welcome to Netgate pfSense Plus 23.01-RELEASE (amd64) on pfSense ***

 WAN (wan)       -> igb1       -> v4/DHCP4: x.x.x.x/20
 LAN (lan)       -> igb0       -> v4: 192.168.0.1/24
 OPT1 (opt1)     -> em0        -> 

 0) Logout (SSH only)                  9) pfTop
 1) Assign Interfaces                 10) Filter Logs
 2) Set interface(s) IP address       11) Restart webConfigurator
 3) Reset webConfigurator password    12) PHP shell + Netgate pfSense Plus tools
 4) Reset to factory defaults         13) Update from console
 5) Reboot system                     14) Disable Secure Shell (sshd)
 6) Halt system                       15) Restore recent configuration
 7) Ping host                         16) Restart PHP-FPM
 8) Shell

Netgate pfSense Plus - Serial: xxxx - Netgate Device ID: xxxx

*** Welcome to Netgate pfSense Plus 23.01-RELEASE (amd64) on pfSense ***

 WAN (wan)       -> igb1       -> v4/DHCP4: x.x.x.x/20
 LAN (lan)       -> igb0       -> v4: 192.168.0.1/24
 OPT1 (opt1)     -> em0        -> 

 0) Logout (SSH only)                  9) pfTop
 1) Assign Interfaces                 10) Filter Logs
 2) Set interface(s) IP address       11) Restart webConfigurator
 3) Reset webConfigurator password    12) PHP shell + Netgate pfSense Plus tools
 4) Reset to factory defaults         13) Update from console
 5) Reboot system                     14) Disable Secure Shell (sshd)
 6) Halt system                       15) Restore recent configuration
 7) Ping host                         16) Restart PHP-FPM
 8) Shell

0END

C:\Users\x>`

**Comment by crschnick** [comment]
This is probably the same problem as in https://github.com/xpipe-io/xpipe/issues/89, but it was only fixed for opnsense as I only had access to that. I should be able to adapt the fix to support pfsense as well

**Comment by crschnick** [comment]
Can you try whether the latest PTB version works for you: https://github.com/xpipe-io/xpipe-ptb ? That would save me a lot of time setting up pfsense just to test this.

**Comment by crschnick** [comment]
This should be fixed in the latest release

**Comment by paulharman** [comment]
Sorry I didn't see your message before.  Unfortunately it doesn't seem to be working, same issue on 1.7.13


Shell did not start up in time as we did not receive any response. In case the shell is just slow, you can try increasing the timeout value in the settings.

Netgate pfSense Plus - Serial: xxxx - Netgate Device ID: xxxxx

*** Welcome to Netgate pfSense Plus 23.01-RELEASE (amd64) on pfSense ***

 WAN (wan)       -> igb1       -> v4/DHCP4: x.x.x.x/20
 LAN (lan)       -> igb0       -> v4: 192.168.0.1/24
 OPT1 (opt1)     -> em0        -> 

 0) Logout (SSH only)                  9) pfTop
 1) Assign Interfaces                 10) Filter Logs
 2) Set interface(s) IP address       11) Restart webConfigurator
 3) Reset webConfigurator password    12) PHP shell + Netgate pfSense Plus tools
 4) Reset to factory defaults         13) Update from console
 5) Reboot system                     14) Disable Secure Shell (sshd)
 6) Halt system                       15) Restore recent configuration
 7) Ping host                         16) Restart PHP-FPM
 8) Shell

**Comment by crschnick** [comment]
Can you start xpipe in debug mode and send me the output when it connects somehow? You can find that under Settings -> Troubleshoot

**Comment by paulharman** [comment]
No problem.  Where is the output saved please?

**Comment by crschnick** [comment]
It is only printed in the terminal as it might contain sensitive information, so it's not stored anywhere.

**Comment by paulharman** [comment]
Sorry what terminal? I am on windows

**Comment by crschnick** [comment]
If you successfully launch it in debug mode, there should be a terminal window opening up, e.g. cmd or windows terminal depending on what you use.

**Comment by crschnick** [comment]
Closing this as completed

**Comment by Googlepuss** [comment]
not certain if this was addressed or closed for lack of response. I have this same issue. debug log is below. I set this up as Simple ssh in xpipe, and i can confirm running local shell in tabby will launch ssh session as expected. 

14:49:47:102 - trace: Writing exec script {
    file=/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh
    content=
    | #!/usr/bin/env zsh
    | unset ZDOTDIR
    | while true
    | do
    | '/Applications/XPipe.app/Contents/MacOS/xpipe' terminal-wait --port 21721 --request c6b97ebe-b7d5-4531-bb9e-83766b2232ef
    | r=$?
    | if [ $r -ne 0 ]; then
    |   echo "Press [r] to restart or any other key to close this session ..."
    |   read -rsk;export key="$REPLY"
    |   if [ "$key" = "r" ]; then
    |     continue
    |   fi
    |   exit $r
    | fi
    | 
    | SCRIPT="$('/Applications/XPipe.app/Contents/MacOS/xpipe' terminal-launch --port 21721 --request c6b97ebe-b7d5-4531-bb9e-83766b2232ef)"
    | r=$?
    | if [ $r -ne 0 ]; then
    |   echo "Press [r] to restart or any other key to close this session ..."
    |   read -rsk;export key="$REPLY"
    |   if [ "$key" = "r" ]; then
    |     continue
    |   fi
    |   exit $r
    | fi
    | 
    | if [ -f "$SCRIPT" ]; then
    |   . "$SCRIPT"
    |   echo "Press [r] to restart or any other key to close this session ..."
    |   read -rsk;export key="$REPLY"
    |   if [ "$key" = "r" ]; then
    |     continue
    |   fi
    |   exit 0
    | else
    |   echo "Press enter to close this session ..."; read -sk
    |   exit 1
    | fi
    | done
    | 
}
14:49:47:102 - info: Starting command execution ... {
    baseCommand=cat << '87f47672-7530-4db0-a6af-547cb04410e7' > '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh' && chmod +x '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh'
    shellType=zsh
    elevated=false
    workingDirectory=null
}
14:49:47:103 - trace: Writing line {
    line= echo d${INVALID}8b76f2f-9642-4a73-b829-0c22171bc10c;echo '9fe8de00-58a6-4c61-832a-a08218748a85' 1>&2;cat << '87f47672-7530-4db0-a6af-547cb04410e7' > '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh' && chmod +x '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh';echo 'f15cb6a5-af61-4afb-abc9-56dc81b937d5'E${INVALID}XIT${?}END;echo '008ebca4-cef4-43e7-b027-05357e2e0912' 1>&2
}
14:49:47:103 - trace: Writing line {
    line=
    | #!/usr/bin/env zsh
    | unset ZDOTDIR
    | while true
    | do
    | '/Applications/XPipe.app/Contents/MacOS/xpipe' terminal-wait --port 21721 --request c6b97ebe-b7d5-4531-bb9e-83766b2232ef
    | r=$?
    | if [ $r -ne 0 ]; then
    |   echo "Press [r] to restart or any other key to close this session ..."
    |   read -rsk;export key="$REPLY"
    |   if [ "$key" = "r" ]; then
    |     continue
    |   fi
    |   exit $r
    | fi
    | 
    | SCRIPT="$('/Applications/XPipe.app/Contents/MacOS/xpipe' terminal-launch --port 21721 --request c6b97ebe-b7d5-4531-bb9e-83766b2232ef)"
    | r=$?
    | if [ $r -ne 0 ]; then
    |   echo "Press [r] to restart or any other key to close this session ..."
    |   read -rsk;export key="$REPLY"
    |   if [ "$key" = "r" ]; then
    |     continue
    |   fi
    |   exit $r
    | fi
    | 
    | if [ -f "$SCRIPT" ]; then
    |   . "$SCRIPT"
    |   echo "Press [r] to restart or any other key to close this session ..."
    |   read -rsk;export key="$REPLY"
    |   if [ "$key" = "r" ]; then
    |     continue
    |   fi
    |   exit 0
    | else
    |   echo "Press enter to close this session ..."; read -sk
    |   exit 1
    | fi
    | done
    | 
    | 87f47672-7530-4db0-a6af-547cb04410e7
}
14:49:47:124 - trace: Stdout finished. Reading exit code ...
14:49:47:124 - trace: Stderr finished {
    finishReason=NORMAL_FINISH
}
14:49:47:125 - trace: Read stderr {
    output=
}
14:49:47:125 - info: Command finished {
    command=cat << '87f47672-7530-4db0-a6af-547cb04410e7' > '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh' && chmod +x '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh'
    finishReason=NORMAL_FINISH
    exitCode=0
}
14:49:47:125 - trace: Command state switched to finished as both streams finished
14:49:47:125 - trace: Read stdout {
    output=
}
14:49:47:125 - trace: Closing command normally {
    manageParent=false
}
14:49:47:126 - info: Starting command execution ... {
    baseCommand=open -a "Tabby.app" -n --args run '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh'
    shellType=zsh
    elevated=false
    workingDirectory=null
}
14:49:47:126 - trace: Writing line {
    line= echo 3${INVALID}d6758b6-1fa2-4a82-b657-fb2d26973ba8;echo '32e8e46b-8866-450a-8aaf-ad0c1fdf6b83' 1>&2;open -a "Tabby.app" -n --args run '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh';echo '51f25c2a-0e18-4692-be6a-8989b828bc55'E${INVALID}XIT${?}END;echo '87467a07-7f96-42a3-acce-5449082f9836' 1>&2
}
14:49:47:134 - trace: Starting local shell {
    command=/bin/zsh -l
    id=b849c8e4-f161-42ce-8b87-4ec565dc1f66
}
14:49:47:134 - trace: Writing line {
    line= echo -NoEnumerate "xpipetest"
}
14:49:47:134 - trace: Writing line {
    line=
}
14:49:47:269 - trace: Read: -NoEnumerate xpipetest
14:49:47:269 - trace: Writing line {
    line= printf "shell="; test -f "/proc/$$/cmdline" && (which xargs > /dev/null && which cut > /dev/null && which cat > /dev/null && cat "/proc/$$/cmdline" | xargs -0 | cut -d ' ' -f1) || ps -cp "$$" -o command="" || echo sh
}
14:49:47:271 - trace: Stdout finished. Reading exit code ...
14:49:47:273 - trace: Stderr finished {
    finishReason=NORMAL_FINISH
}
14:49:47:273 - info: Command finished {
    command=open -a "Tabby.app" -n --args run '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/exec-57169911.sh'
    finishReason=NORMAL_FINISH
    exitCode=0
}
14:49:47:273 - trace: Command state switched to finished as both streams finished
14:49:47:273 - trace: Read stdout {
    output=
}
14:49:47:273 - trace: Read stderr {
    output=
}
14:49:47:273 - trace: Closing command normally {
    manageParent=false
}
14:49:47:284 - trace: Read: shell=zsh
14:49:47:285 - trace: Writing line {
    line=export NOCOLOR='true'
}
14:49:47:285 - trace: Writing line {
    line=export TERM='dumb'
}
14:49:47:285 - trace: Writing line {
    line=export LC_ALL='en_US.UTF-8' 2> /dev/null
}
14:49:47:285 - trace: Writing line {
    line=export LANGUAGE='en_US'
}
14:49:47:285 - trace: Writing line {
    line=test -f ~/.zshrc && . ~/.zshrc
}
14:49:47:285 - info: Starting command execution ... {
    baseCommand=test -d '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/'
    shellType=zsh
    elevated=false
    workingDirectory=null
}
14:49:47:285 - trace: Writing line {
    line= echo 7${INVALID}83abc10-c4a2-4246-9462-29d125b88a0b;echo '1057d09c-f687-4e2e-94bd-5a6aebb42621' 1>&2;test -d '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/';echo '5448c1d9-2548-42f9-b530-4f1237421b79'E${INVALID}XIT${?}END;echo '16897de7-d1db-4099-b601-de9d63fa77aa' 1>&2
}
14:49:47:291 - trace: Stderr finished {
    finishReason=NORMAL_FINISH
}
14:49:47:291 - trace: Read stderr {
    output=
}
14:49:47:292 - trace: Stdout finished. Reading exit code ...
14:49:47:292 - info: Command finished {
    command=test -d '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/'
    finishReason=NORMAL_FINISH
    exitCode=0
}
14:49:47:292 - trace: Command state switched to finished as both streams finished
14:49:47:292 - trace: Read stdout {
    output=
}
14:49:47:292 - trace: Closing command normally {
    manageParent=false
}
14:49:47:293 - info: Starting command execution ... {
    baseCommand=test -r '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/' && test -w '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/' && test -x '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/'
    shellType=zsh
    elevated=false
    workingDirectory=null
}
14:49:47:293 - trace: Writing line {
    line= echo 8${INVALID}5422c2b-1cb7-4fa3-b492-ea34bf284d81;echo 'ab214813-d399-4458-8ac6-9cfd73fd121a' 1>&2;test -r '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/' && test -w '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/' && test -x '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/';echo '678d34a6-f271-4bbb-a713-519c4e1fce82'E${INVALID}XIT${?}END;echo 'c862e75b-31a9-49a9-bf88-e01b4c4b7dfd' 1>&2
}
14:49:47:295 - trace: Stdout finished. Reading exit code ...
14:49:47:296 - info: Command finished {
    command=test -r '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/' && test -w '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/' && test -x '/var/folders/rx/twcxd3js5jj93_lkkvyymvsh0000gn/T/'
    finishReason=NORMAL_FINISH
    exitCode=0
}
14:49:47:296 - trace: Read stdout {
    output=
}
14:49:47:298 - trace: Stderr finished {
    finishReason=NORMAL_FINISH
}
14:49:47:298 - trace: Command state switched to finished as both streams finished
14:49:47:298 - trace: Read stderr {
    output=
}
14:49:47:298 - trace: Closing command normally {
    manageParent=false
}
14:49:47:299 - trace: Writing line {
    line=cd '/Users/teague'
}
14:49:47:301 - info: Starting command execution ... {
    baseCommand=which 'ssh-sk-helper'
    shellType=zsh
    elevated=false
    workingDirectory=null
}
14:49:47:302 - trace: Writing line {
    line= echo 6${INVALID}2488b3a-dc29-4dc6-95f7-2cfc84024e45;echo '95d56a36-8128-47b7-8197-70199a2bdb8d' 1>&2;which 'ssh-sk-helper';echo 'ae688a51-2f40-4f46-bafc-d8462e0926a2'E${INVALID}XIT${?}END;echo '364609d3-d5f4-4eb7-81a8-095ad4506706' 1>&2
}
14:49:47:306 - trace: Stdout finished. Reading exit code ...
14:49:47:306 - info: Command finished {
    command=which 'ssh-sk-helper'
    finishReason=NORMAL_FINISH
    exitCode=1
}
14:49:47:306 - trace: Read stdout {
    output=ssh-sk-helper not found
}
14:49:47:307 - trace: Stderr finished {
    finishReason=NORMAL_FINISH
}
14:49:47:307 - trace: Command state switched to finished as both streams finished
14:49:47:307 - trace: Read stderr {
    output=
}
14:49:47:307 - trace: Closing command normally {
    manageParent=false
}
14:49:47:308 - info: Starting command execution ... {
    baseCommand=echo "$DISPLAY"
    shellType=zsh
    elevated=false
    workingDirectory=null
}
14:49:47:308 - trace: Writing line {
    line= echo 0${INVALID}1bb4bbb-15a8-46cb-a211-7f8db0ccd8ac;echo '664eed60-71ce-4a8b-83a0-38869070197b' 1>&2;echo "$DISPLAY";echo 'e5316430-b3a6-4883-9b85-b9adbcfb5489'E${INVALID}XIT${?}END;echo '4cae85fd-9275-490f-91a7-09505e2e978d' 1>&2
}
14:49:47:309 - trace: Stdout finished. Reading exit code ...
14:49:47:309 - info: Command finished {
    command=echo "$DISPLAY"
    finishReason=NORMAL_FINISH
    exitCode=0
}
14:49:47:309 - trace: Read stdout {
    output=
}
14:49:47:309 - trace: Stderr finished {
    finishReason=NORMAL_FINISH
}
14:49:47:310 - trace: Command state switched to finished as both streams finished
14:49:47:310 - trace: Read stderr {
    output=
}
14:49:47:311 - trace: Closing command normally {
    manageParent=false
}
14:49:47:312 - info: Starting sub shell ... {
    manageParent=true
    command=DISPLAY='localhost:0.0' SSH_ASKPASS_REQUIRE='force' SSH_ASKPASS='/Applications/XPipe.app/Contents/MacOS/xpipe' ASKPASS_PREFIX='[ssh]' ASKPASS_REQUEST='98c14745-eeb5-4d78-b775-92efcc0a96ff' ASKPASS_SECRET='80b5f0c1-407c-35f0-8ddc-a98bcee75ad8' BEACON_PORT='21721' ssh -oNumberOfPasswordPrompts=3 "xxxx@172.xx.xx.xx" -p 22 -oStrictHostKeyChecking=accept-new -oRemoteCommand=none -oNoHostAuthenticationForLocalhost=yes -T -oIdentitiesOnly=yes
    elevated=false
}
14:49:47:312 - trace: Executing sub shell command... {
    command=DISPLAY='localhost:0.0' SSH_ASKPASS_REQUIRE='force' SSH_ASKPASS='/Applications/XPipe.app/Contents/MacOS/xpipe' ASKPASS_PREFIX='[ssh]' ASKPASS_REQUEST='98c14745-eeb5-4d78-b775-92efcc0a96ff' ASKPASS_SECRET='80b5f0c1-407c-35f0-8ddc-a98bcee75ad8' BEACON_PORT='21721' ssh -oNumberOfPasswordPrompts=3 "xxxx@172.xx.xx.xx" -p 22 -oStrictHostKeyChecking=accept-new -oRemoteCommand=none -oNoHostAuthenticationForLocalhost=yes -T -oIdentitiesOnly=yes
}
14:49:47:312 - trace: Writing line {
    line= echo 6${INVALID}a111f27-13c4-40e7-8b3d-df11f44270f6;echo '64d093d4-fb58-4d46-a7aa-b79f5482d2f5' 1>&2;DISPLAY='localhost:0.0' SSH_ASKPASS_REQUIRE='force' SSH_ASKPASS='/Applications/XPipe.app/Contents/MacOS/xpipe' ASKPASS_PREFIX='[ssh]' ASKPASS_REQUEST='98c14745-eeb5-4d78-b775-92efcc0a96ff' ASKPASS_SECRET='80b5f0c1-407c-35f0-8ddc-a98bcee75ad8' BEACON_PORT='21721' ssh -oNumberOfPasswordPrompts=3 "xxxx@172.xx.xx.xx -p 22 -oStrictHostKeyChecking=accept-new -oRemoteCommand=none -oNoHostAuthenticationForLocalhost=yes -T -oIdentitiesOnly=yes;echo '98c14745-eeb5-4d78-b775-92efcc0a96ff'E${INVALID}XIT${?}END;echo '98c14745-eeb5-4d78-b775-92efcc0a96ff' 1>&2
}
14:49:47:315 - trace: Read: 6a111f27-13c4-40e7-8b3d-df11f44270f6
14:49:47:315 - trace: Read: 
14:49:47:316 - trace: Read: 64d093d4-fb58-4d46-a7aa-b79f5482d2f5
14:49:47:316 - trace: Writing line {
    line= echo -NoEnumerate "xpipetest"
}
14:49:47:316 - trace: Writing line {
    line=
}
14:49:49:518 - trace: Parsed raw request:
{
  "auth" : {
    "type" : "Local",
    "authFileContent" : "0354fd5f-6ed0-48d4-b390-85f412c3d33f"
  },
  "client" : {
    "type" : "Cli"
  }
}
14:49:49:518 - trace: Parsed raw request:
{
  "auth" : {
    "type" : "Local",
    "authFileContent" : "0354fd5f-6ed0-48d4-b390-85f412c3d33f"
  },
  "client" : {
    "type" : "Cli"
  }
}
14:49:49:519 - trace: Parsed request object:
HandshakeExchange.Request(auth=BeaconAuthMethod.Local(authFileContent=0354fd5f-6ed0-48d4-b390-85f412c3d33f), client=BeaconClientInformation.Cli())
14:49:49:519 - trace: Parsed request object:
HandshakeExchange.Request(auth=BeaconAuthMethod.Local(authFileContent=0354fd5f-6ed0-48d4-b390-85f412c3d33f), client=BeaconClientInformation.Cli())
14:49:49:519 - trace: Sending response:
HandshakeExchange.Response(sessionToken=148a6a19-edad-46d1-a744-a286a91f0eec)
14:49:49:519 - trace: Sending response:
HandshakeExchange.Response(sessionToken=09572f58-8222-4c1c-9ae1-9bb4502eb17c)
14:49:49:520 - trace: Sending raw response:
{
  "sessionToken" : "148a6a19-edad-46d1-a744-a286a91f0eec"
}
14:49:49:520 - trace: Sending raw response:
{
  "sessionToken" : "09572f58-8222-4c1c-9ae1-9bb4502eb17c"
}
14:49:49:541 - trace: Parsed raw request:
{
  "request" : "c6b97ebe-b7d5-4531-bb9e-83766b2232ef",
  "pid" : 45843
}
14:49:49:541 - trace: Parsed raw request:
{
  "pid" : 45842,
  "secretId" : "80b5f0c1-407c-35f0-8ddc-a98bcee75ad8",
  "request" : "98c14745-eeb5-4d78-b775-92efcc0a96ff",
  "prompt" : "[ssh] (xxxx@172.xx.xx.xx) Password for xxxx@xxxxxxxx:"
}
14:49:49:541 - trace: Parsed request object:
TerminalWaitExchange.Request(request=c6b97ebe-b7d5-4531-bb9e-83766b2232ef, pid=45843)
14:49:49:541 - trace: Parsed request object:
AskpassExchange.Request(pid=45842, secretId=80b5f0c1-407c-35f0-8ddc-a98bcee75ad8, request=98c14745-eeb5-4d78-b775-92efcc0a96ff, prompt=[ssh] (xxxx@172.xx.xx.xx) Password for xxxx@xxxxxxxx:)
14:49:49:542 - trace: Shell session opened {
    pid=45841
}
14:49:49:542 - trace: Terminal session opened {
    pid=44195
    exec=/Applications/Tabby.app/Contents/MacOS/Tabby
}
14:49:49:542 - trace: Terminal instance opened {
    terminalPid=44195
}
14:49:49:542 - trace: Sending response:
AskpassExchange.Response(value=<in place secret>)
14:49:49:547 - trace: Sending raw response:
{
  "value" : "<secret>"
}
14:49:50:015 - trace: Read: *** *** *** *** *** ***
14:49:50:025 - trace: Read: This script must be run from an interactive terminal and not through FastCGI
14:49:50:035 - trace: Read: Password MUST be changed.
14:49:50:045 - trace: Read: If the password has already been changed in the GUI, press ctrl-c to cancel.
14:49:50:055 - trace: Read: *** *** *** *** *** ***
14:49:50:185 - trace: Read: This script must be run from an interactive terminal and not through FastCGI
14:49:50:188 - trace: Read: stty: stdin isn't a terminal
14:49:50:622 - trace: Read: 98c14745-eeb5-4d78-b775-92efcc0a96ff
14:49:50:622 - trace: Read: 98c14745-eeb5-4d78-b775-92efcc0a96ffEXIT0END
14:49:51:623 - trace: Read: 
14:49:51:624 - trace: Killing sub shell {
    command=DISPLAY='localhost:0.0' SSH_ASKPASS_REQUIRE='force' SSH_ASKPASS='/Applications/XPipe.app/Contents/MacOS/xpipe' ASKPASS_PREFIX='[ssh]' ASKPASS_REQUEST='98c14745-eeb5-4d78-b775-92efcc0a96ff' ASKPASS_SECRET='80b5f0c1-407c-35f0-8ddc-a98bcee75ad8' BEACON_PORT='21721' ssh -oNumberOfPasswordPrompts=3 "xxxx@172.xx.xx.xx" -p 22 -oStrictHostKeyChecking=accept-new -oRemoteCommand=none -oNoHostAuthenticationForLocalhost=yes -T -oIdentitiesOnly=yes
}
14:49:51:624 - trace: Killing local shell ...
14:49:51:626 - trace: Killing sub shell {
    command=DISPLAY='localhost:0.0' SSH_ASKPASS_REQUIRE='force' SSH_ASKPASS='/Applications/XPipe.app/Contents/MacOS/xpipe' ASKPASS_PREFIX='[ssh]' ASKPASS_REQUEST='98c14745-eeb5-4d78-b775-92efcc0a96ff' ASKPASS_SECRET='80b5f0c1-407c-35f0-8ddc-a98bcee75ad8' BEACON_PORT='21721' ssh -oNumberOfPasswordPrompts=3 "xxxx@172.xx.xx.xx" -p 22 -oStrictHostKeyChecking=accept-new -oRemoteCommand=none -oNoHostAuthenticationForLocalhost=yes -T -oIdentitiesOnly=yes
}
14:49:51:630 - error: Shell opener command was unsuccessful:
*** *** *** *** *** ***
This script must be run from an interactive terminal and not through FastCGI
Password MUST be changed.
If the password has already been changed in the GUI, press ctrl-c to cancel.
*** *** *** *** *** ***
This script must be run from an interactive terminal and not through FastCGI

stty: stdin isn't a terminal
io.xpipe.core.process.ProcessOutputException: Shell opener command was unsuccessful:
*** *** *** *** *** ***
This script must be run from an interactive terminal and not through FastCGI
Password MUST be changed.
If the password has already been changed in the GUI, press ctrl-c to cancel.
*** *** *** *** *** ***
This script must be run from an interactive terminal and not through FastCGI

stty: stdin isn't a terminal
        at io.xpipe.core.process.ProcessOutputException.withPrefix(ProcessOutputException.java:29)
        at io.xpipe.ext.proc.util.f.bL(SourceFile:182)
        at io.xpipe.ext.proc.m.p(SourceFile:2642)
        at io.xpipe.ext.proc.ssh.r.p(SourceFile:78)
        at io.xpipe.ext.proc.m.D(SourceFile:541)
        at io.xpipe.ext.proc.m.start(SourceFile:255)
        at io.xpipe.ext.proc.m.a(SourceFile:362)
        at io.xpipe.ext.proc.m.prepareIntermediateTerminalOpen(SourceFile:271)
        at io.xpipe.ext.proc.i.prepareTerminalOpen(SourceFile:74)
        at io.xpipe.app.terminal.TerminalLaunchRequest.setupRequest(TerminalLaunchRequest.java:97)
        at io.xpipe.app.terminal.TerminalLaunchRequest.lambda$setupRequestAsync$0(TerminalLaunchRequest.java:64)
        at io.xpipe.app.util.ThreadHelper.lambda$wrap$0(ThreadHelper.java:19)
        at java.lang.Thread.run(Thread.java:1570)

**Comment by crschnick** [comment]
So, this issue was originally closed because it was fixed, at least for the original case.

I don't have pfsense here, so I can't reproduce it myself. But looking at the output, that is quite different from the original post. If you connect to that system manually, how do you run a command in a shell there? Usually there should be a menu where you select something like `8) Shell` on login.

**Comment by crschnick** [comment]
Where does this output come from
```
This script must be run from an interactive terminal and not through FastCGI
Password MUST be changed.
If the password has already been changed in the GUI, press ctrl-c to cancel.
```

What if you run `ssh -T user@host`

**Comment by Googlepuss** [comment]
That is the text is parented as and error pop up in xpipe, and at the bottom of the debug log I posted.I will be back at a laptop in 30min or so and will let you how -T acts.Teague Johnson501.765.9423 /// “Machines should work. People should think.”- IBMOn Jan 16, 2025, at 7:04 AM, Christopher Schnick ***@***.***> wrote:﻿
Where does this output come from
This script must be run from an interactive terminal and not through FastCGI
Password MUST be changed.
If the password has already been changed in the GUI, press ctrl-c to cancel.

What if you run ssh -T ***@***.***

**Comment by crschnick** [comment]
But the error must come from pfsense, that is where xpipe is reading it from. It might be the -T switch, so we will see what happens. You can also test whether XPipe 14, which has just been released, makes a difference.

**Comment by Googlepuss** [comment]
that did it. 

 ~ % ssh -T user@172.x.x.x
(user@172.x.x.x) Password for xxx@xxxxxxxx:
*** *** *** *** *** ***
This script must be run from an interactive terminal and not through FastCGI
Password MUST be changed.
If the password has already been changed in the GUI, press ctrl-c to cancel.
*** *** *** *** *** ***
This script must be run from an interactive terminal and not through FastCGI
stty: stdin isn't a terminal

**Comment by crschnick** [comment]
Hmm, interesting. What is the password warning about?

**Comment by Googlepuss** [comment]
I can open sessions with -T on other devices (not pfsense) from this device, is that what you mean?

~ % ssh -T user@xxxxxxxxxxx
user@sxxxxxxxxx's password: 
Welcome to Ubuntu 22.04.5 LTS (GNU/Linux 5.15.0-130-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/pro

 System information as of Thu Jan 16 02:00:45 PM UTC 2025

  System load:  0.61                Processes:             199
  Usage of /:   12.5% of 488.32GB   Users logged in:       1
  Memory usage: 26%                 IPv4 address for eth0: 172.2.x.x
  Swap usage:   10%

**Comment by crschnick** [comment]
I'm focusing on the pfsense system. If it's not possible to open a shell with -T, then I can fall back to a limited interaction mode where you can still launch the connection in the terminal, but other features such as the file browser do not work.

**Comment by Googlepuss** [comment]
I got it worked out. Thanks for the feedback. root user on pfsense gets landed on that launch page with numeric options, which is throwing that error. I created a new user and added them to admins group on pfsense and it launches directly into the shell. so I ran it with -T and get the prompt below. 

~ % ssh -T tusere@172.2.x.x
(user@172.x.x.x) Password for xxxx@xxxxxxxxxxx:
Warning: no access to tty (Bad file descriptor).
Thus no job control in this shell.

I went ahead and tried to add it to xpipe, and was successful. I have access to the folder structure as well. TY.
